SU2_CFD/src/solvers/CFEASolver.cpp

```diff
@@ -879,6 +912,8 @@ void CFEASolver::Compute_StiffMatrix(CGeometry *geometry, CNumerics **numerics,
         /*--- Update residual and stiffness matrix with contributions from the element. ---*/
         for (iNode = 0; iNode < nNodes; iNode++) {
 
+          if (LockStrategy) omp_set_lock(&UpdateLocks[indexNode[iNode]]);
```

Comment:
In the loops over the points associated with an element, that follow the computation of element stiffness matrices (for example) we protected against concurrent accesses by setting (or acquiring) a lock for the point before writing to shared memory locations. Once done we unset the lock.